### ISSUE TITLE
Missing levels in level selection (endless mode)

### ISSUE THREAD

**Issue by CrsiX** [ochadenas/cpudefense#113]
As seen in the two screenshots below, the list of levels in the endless mode selection stops at 28 (see the scroll bar), even though I am currently playing on level 32 as seen in the main menu (and [here](https://github.com/ochadenas/cpudefense/assets/47722349/385f36e5-3edd-4919-8871-fcdc23f9d9fa)).
If needed I can provide the safe file, if you tell me where to look for it on a rooted device.
Closing the app and opening doesn't resolve the issue, it's also not resolved by going into the level and then back to the main menu.

![endless-stored-levels](https://github.com/ochadenas/cpudefense/assets/47722349/386f8397-74d7-4ef8-8d13-afbbc283233a)

![main-menu-level-32](https://github.com/ochadenas/cpudefense/assets/47722349/5b9011c8-e436-4e6d-bac3-0eb0e5c66617)

**Comment by ochadenas** [comment]
fixed in 1.30.

**Comment by ochadenas** [comment]
> If needed I can provide the safe file, if you tell me where to look for it on a rooted device. 

No, it's ok, I have found the bug.

**Comment by CrsiX** [comment]
Thank you very much. That was a quick fix 👍🏼

**Comment by Kutabu** [comment]
@ochadenas : The bug is not finally fixed in version 1.30

The level selection of the endless levels is still limited (don't scroll down any further). Due to adding of level 29, the selection of endless levels is also limited to this point. This doesn't matter for new players because they have to play all levels anyway until endless mode is unlocked, but the bug will still occur in the future when more standard levels are added.

**Comment by Kutabu** [comment]
Found the bug:
app/src/main/java/com/example/cpudefense/LevelSelectActivity.kt
Line 154 in commit dd4e316

The "won"-condition was reseted with adding a new level.
Is `  && (stageSummary[highestLevelInList]?.won == true))` even necessary? Or is this the way to prevent the levels from being displayed until you have finished the standard levelling?

**Comment by Kutabu** [comment]
I'm sorry for cluttering up this issue. After I didn't do anything to reproduce the error, I played the new levels during my lunch break.
In endless mode, the level selection is still limited to 29. So the bug still exists. It must have something to do with the stage.summary not being called correctly. When I click on the level on the main screen, it also tells me that I have reached level 53 in "normal" mode. 

![grafik](https://github.com/ochadenas/cpudefense/assets/125028940/7d81aef9-a110-43ec-b1f2-756d2d801ebf)
